sw/airborne/modules/computer_vision/viewvideo.c

```diff
@@ -186,22 +206,73 @@ struct image_t *viewvideo_function(struct image_t *img)
       // (1 = 1/90000 s) which is probably stupid but is actually working.
     }
 #endif
-
   }
 
   // Free all buffers
   image_free(&img_jpeg);
-  image_free(&img_small);
-  return NULL; // No new images were created
 }
 
+
+void *viewvideo_thread(void *args);
+void *viewvideo_thread(void *args) {
+  // Time-keeping variables
+  struct timespec time_now, time_prev;
+  clock_gettime(CLOCK_MONOTONIC, &time_prev);
+  int32_t delta = 0;
+  uint32_t fps_period = 1000000 / VIEWVIDEO_FPS;
+
+  // Request new image from video thread
+  pthread_mutex_lock(&img_mutex);
```

Comment:
Right, was badly phrased and I didn't have a close enough look (thought you were unlocking the mutex inside the while).
The above would still be the common usage pattern if you don't want/need to lock the mutex for the whole time, e.g if not the whole part is a critical section.